docs/x509.rst

```diff
@@ -166,6 +166,143 @@ X.509 Certificate Object
             >>> cert.not_valid_after
             datetime.datetime(2030, 12, 31, 8, 30)
 
+    .. attribute:: issuer
+
+        .. versionadded:: 0.8
+
+        :type: :class:`Name`
+
+        The :class:`Name` of the issuer.
+
+    .. attribute:: subject
+
+        .. versionadded:: 0.8
+
+        :type: :class:`Name`
+
+        The :class:`Name` of the subject.
+
+
+.. class:: Name
+
+    .. versionadded:: 0.8
+
+    An X509 Name is an ordered list of attributes. The entire list can be
+    obtained with :attr:`attributes` or you can use the helper properties to
+    obtain the specific type you want. Names are sometimes represented as a
+    slash or comma delimited string (e.g. ``/CN=mydomain.com/O=My Org/C=US``).
+
+    .. attribute:: attributes
+
+        :type: :class:`list`
+
+        A list of all the :class:`NameAttribute` objects.
+
+        .. doctest::
+
+            >>> len(cert.subject.attributes)
+            3
+
+    .. attribute:: country_name
+
+        :type: :class:`list`
+
+        A list of country name :class:`NameAttribute` objects.
+
+        .. doctest::
+
+            >>> cert.subject.country_name == [
```

Comment:
Oh. Something to think about
On Feb 11, 2015 7:27 PM, "Paul Kehrer" <notifications@github.com> wrote:

> In docs/x509.rst
> <https://github.com/pyca/cryptography/pull/1612#discussion_r24548466>:
>
> > +        A list of all the :class:`NameAttribute` objects.
> > +
> > +        .. doctest::
> > +
> > +            >>> len(cert.subject.attributes)
> > +            3
> > +
> > +    .. attribute:: country_name
> > +
> > +        :type: :class:`list`
> > +
> > +        A list of country name :class:`NameAttribute` objects.
> > +
> > +        .. doctest::
> > +
> > +            >>> cert.subject.country_name == [
>
> We don't have a repr for NameAttribute, just ObjectIdentifier. We could
> add one to NameAttribute, although we'd either need to return it as unicode
> or decode all values to ascii in the repr.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/1612/files#r24548466>.
>